interfaces/cython/cantera/onedim.pyx

```diff
@@ -875,6 +875,15 @@ cdef class Sim1D:
             dom.set_steady_tolerances(default=(1e-4, 1e-9))
             dom.set_transient_tolerances(default=(1e-4, 1e-11))
 
+        # Do initial steps without Soret diffusion.
+        # Has to be done before the multicomponent check because setting
+        # the transport model to 'Mix' with soret_enabled throws an error
+        solve_soret = any(getattr(dom, 'soret_enabled', False) for dom in self.domains)
```

Comment:
What about writing
```python
soret_doms = [dom for dom in self.domains if getattr(dom, 'soret_enabled', False)]
```
then just using this list below, to spare the `isinstance` checks (and the `if` as well, for that matter)?

Reply:
OK, this should work since an empty list will evaluate to False for the checks further down